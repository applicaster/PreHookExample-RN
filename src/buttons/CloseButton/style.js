import {
  StyleSheet,
} from 'react-native';

const closeButtonSize = 38;

export const styles = StyleSheet.create({
  closeButton: {
    height: closeButtonSize,
    width: closeButtonSize,
  },
});
