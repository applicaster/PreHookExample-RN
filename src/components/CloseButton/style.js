import {
  StyleSheet,
} from 'react-native';

const closeButtonSize = 25;

export const styles = StyleSheet.create({
  closeButton: {
    tintColor: '#FFFFFF',
    height: closeButtonSize,
    width: closeButtonSize,
    marginLeft: 5,
  },
  modalCloseButton: {
    position: 'absolute',
    top: 30,
    left: 5,
    zIndex: 5,
  },
});
