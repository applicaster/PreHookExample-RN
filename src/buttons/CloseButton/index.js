import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import { styles } from './style';
import { CLOSE_ROUND_BUTTON } from '../../icons';

class CloseButton extends Component {
  render() {
    const { style: buttonContainerStyles, onPress, tintColor } = this.props;
    const { textColor = '#FFFFFF' } = this.context;
    const closeButtonColor = { tintColor: tintColor || `${textColor}99` };

    return (
      <TouchableWithoutFeedback onPress={onPress}>
        <View style={buttonContainerStyles}>
          <Image style={[styles.closeButton, closeButtonColor]} source={{ uri: CLOSE_ROUND_BUTTON }} />
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

CloseButton.propTypes = {
  onPress: PropTypes.func,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
  tintColor: PropTypes.string,
};

CloseButton.contextTypes = {
  textColor: PropTypes.string,
};

export default CloseButton;
