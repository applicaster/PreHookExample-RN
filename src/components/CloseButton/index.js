import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Image,
  TouchableWithoutFeedback,
} from 'react-native';
import { styles } from './style';

const CLOSE_BUTTON = 'feed_rn_close_button';
class CloseButton extends Component {
  render() {
    const { onPress, isForModal } = this.props;
    const { textColor = '#FFFFFF' } = this.context;
    const closeButtonColor = { tintColor: `${textColor}99` };
    const isForModalStyles = (isForModal) ? styles.modalCloseButton : null;

    return (
      <TouchableWithoutFeedback onPress={onPress}>
        <Image style={[styles.closeButton, isForModalStyles, closeButtonColor]} source={{ uri: CLOSE_BUTTON }} />
      </TouchableWithoutFeedback>
    );
  }
}

CloseButton.propTypes = {
  onPress: PropTypes.func,
  isForModal: PropTypes.bool,
};

CloseButton.contextTypes = {
  textColor: PropTypes.string,
};

export default CloseButton;
