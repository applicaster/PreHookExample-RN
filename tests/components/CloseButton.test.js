import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CloseButton from '../../src/buttons/CloseButton';

describe('CloseButton', () => {
  test('component renders correctly when it is not for modal', () => {
    const html = renderer.create(
      <CloseButton onPress={() => {}} isForModal={false} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when it is for modal', () => {
    const html = renderer.create(
      <CloseButton onPress={() => {}} isForModal />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
