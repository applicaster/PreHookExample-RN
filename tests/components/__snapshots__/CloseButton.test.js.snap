// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton component renders correctly when it is for modal 1`] = `
<Image
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  source={
    Object {
      "uri": "feed_rn_close_button",
    }
  }
  style={
    Array [
      Object {
        "height": 25,
        "marginLeft": 5,
        "tintColor": "#FFFFFF",
        "width": 25,
      },
      Object {
        "left": 5,
        "position": "absolute",
        "top": 30,
        "zIndex": 5,
      },
      Object {
        "tintColor": "#FFFFFF99",
      },
    ]
  }
  testID={undefined}
/>
`;

exports[`CloseButton component renders correctly when it is not for modal 1`] = `
<Image
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  source={
    Object {
      "uri": "feed_rn_close_button",
    }
  }
  style={
    Array [
      Object {
        "height": 25,
        "marginLeft": 5,
        "tintColor": "#FFFFFF",
        "width": 25,
      },
      null,
      Object {
        "tintColor": "#FFFFFF99",
      },
    ]
  }
  testID={undefined}
/>
`;
